demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1613,6 +1622,15 @@ export class DemoMeetingApp
       this.enableUnifiedPlanForChromiumBasedBrowsers
     ) {
       filters = filters.concat(VIDEO_FILTERS);
+      if (platformCanSupportBodyPixWithoutDegradation()) {
+        try {
+          await loadBodyPixDependency(this.loadingBodyPixDependencyTimeoutMs);
```

Comment:
This is probably wrong for a couple of reasons.

Firstly, this whole promise chain is `await`ed — you're basically blocking page load. Try it out: put

```
setTimeout(resolve, 20000);
```

in `loadBodyPixDependency`, and see how long it takes to load a meeting. This is a bad pattern for customers to follow.

Secondly, each time the user passes through the authentication flow, you will re-load these files. Try it: hit the 'Leave' button in the top right of a meeting, and join a second meeting.

You will need to switch to using `.then` here to fix the first issue, and use a single, persistent load promise to fix the second issue.

Reply:
Sorry I am not familiar with UI best practice. 
So in this case, the dependencies should be loaded before the filter item is added to the drop down menu, that's why it's await-ed. With `then`, we don't suspend but (re)populate the filter list in the body of `then`, kind of like 
```
private async populateVideoFilterInputList (): void Promise<void> {
        loadDepedencies.then(() => {
           add segmentation filter to list and populate filter list
        });
        populate other filters.
}
```
So user could see a drop down menu without a segmentation filter but click a second time it show up ?

Reply:
You would just add 'Segmentation' to the list in the `.then` block, correct. You will also need a `.catch` block to handle failure. This will allow the rest of the initialization work to occur in parallel.

Reply:
Thanks. For the second issue, https://github.com/aws/amazon-chime-sdk-js/blob/master/demos/browser/app/meetingV2/meetingV2.ts#L2245 page is redirected here when session is stopped. 
It makes sense in general to have a global loading of dependency, but it seems like it does nothing in our demo because page is refreshed when user leaves. Do you mind if I leave persistent loading out for now ?

Reply:
It's preferable to show common best practices if you can.

This should be as simple as somewhere at the top of the file:

```
const bodyPixLoad: Promise<void>;
```

and down here:

```
if (!this.bodyPixLoad) {
  this.bodyPixLoad = loadBodyPixDependency(…);
}
this.bodyPixLoad.then(() => {
  // Add to menu.
}).catch(e => {
  this.logger.warn('Could not load bodyPix:' + e);
});
```

Reply:
When page is re-directed, new demo object is created. I tried moving this out of demo class but it seems like the variable is not retained.

Reply:
Yes, you will get a new page load for the second meeting, so anything you store won't be retained. The point of doing this work is that the demo app doesn't have to work this way, and many customer apps won't — they will join multiple calls within the same JS environment and page scope.

If you want to test that you got it right, call `populateVideoFilterInputList` twice during page load. You should only get one fetch for each BodyPix resource.

Reply:
that makes sense. Changed.